therapy/query.py

```diff
@@ -205,70 +210,32 @@ def check_concept_id(self,
         return (resp, sources)
 
-    def check_label_tn(self,
-                       query: str,
-                       resp: Dict,
-                       sources: Set[str]) -> (Dict, Set):
-        """Check query for label/trade name match.
-
-        Args:
-            query: search string
-            resp: in-progress response object to return to client
-            sources: remaining unmatched sources
-
-        Returns:
-            Tuple with updated resp object and updated unmatched sources set
-        """
-        filter_exp = Key('label_and_type').eq(f'{query}##label')
-        items = []
-        try:
-            db_response = self.db.therapies.query(
-                KeyConditionExpression=filter_exp
-            )
-            items = db_response['Items'][:]
-        except ClientError as e:
-            print(e.response['Error']['Message'])
-
-        filter_exp = Key('label_and_type').eq(f'{query}##trade_name')
-        try:
-            db_response = self.db.therapies.query(
-                KeyConditionExpression=filter_exp
-            )
-            items += db_response['Items'][:]
-        except ClientError as e:
-            print(e.response['Error']['Message'])
-
-        if len(items) > 0:
-            concept_ids = {i['concept_id'] for i in items}
-            (resp, matched_srcs) = self.fetch_records(resp, concept_ids,
-                                                      MatchType.PRIMARY_LABEL)
-            sources = sources - matched_srcs
-        return (resp, sources)
-
-    def check_alias(self,
-                    query: str,
-                    resp: Dict,
-                    sources: Set) -> (Dict, Set):
-        """Check query for alias match.
-
-        Args:
-            query: search string
-            resp: in-progress response object to return to client
-            sources: remaining unmatched sources
-
-        Returns:
-            Tuple with updated resp object and updated unmatched sources set
+    def check_match_type(self,
+                         query: str,
+                         resp: Dict,
+                         sources: Set[str],
+                         match: str) -> (Dict, Set):
+        """Check query for selected match type.
+        :param str query: search string
+        :param Dict resp: in-progress response object to return to client
+        :param Set[str] sources: remaining unmatched sources
+        :param str match: Match type name
+        :return: Tuple with updated resp object and updated set of unmatched
+                 sources
         """
-        filter_exp = Key('label_and_type').eq(f'{query}##alias')
+        filter_exp = Key('label_and_type').eq(f'{query}##{match}')
         try:
             db_response = self.db.therapies.query(
                 KeyConditionExpression=filter_exp
             )
             if 'Items' in db_response.keys():
+                if match == 'label':
+                    match = 'primary_label'
```

Comment:
I hadn't thought about the separation between verbiage in the schema ("primary label") and how it's stored in the db ("label##") before. It's ultimately NBD, but it'd be nice if it was consistent one way or the other so that we don't have to perform these manual condition checks/string replacements (and it's one more place that we have to change if we ever make adjustments to the match type schema). Low-impact in the end, but maybe something to return to someday.

Reply:
Do you think it's better to use 'primary label' or 'label'?

Reply:
I think the change to 'label' would be the best option.

Reply:
I like "label" (brevity!) but I think "primary_label" came from @ahwagner? I'm irresolute.